### PR TITLE
Refactor: Improve friend requests.

### LINKED ISSUE #215: Improve Friend Requests Dialog to not close.
This issue aims to fix the Dialog closing after the user accepts a friend request.

### PR DESCRIPTION
**Pull request by 1andreea26**
## Summary

This PR fixes the issue in which, after the user accepts a friend request, the dialog closed and they would have to open it again to visualize the friend requests. Now, the user views a list, which if empty displays a text saying there are no pending requests, otherwise displaying the requests. After approving a request, the request disappears but the list still stays open. If there are more than 3 friend requests, the user can scroll below to approve/deny them.
Moreover, documentation has been updated for the  ProfileScreen and UserViewModel.

## Changes

- **Screens/UI:** Changes to the friend request list in ProfileScreen.
- **Bug Fixes/Improvements:** Improves user experience and addresses #215 .

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [X] This PR resolves #215.
- [X] Tests have been added for new functionality.
- [X] Screenshots or UI changes have been attached (if applicable).
- [X] Documentation has been updated (if applicable).

##  Testing

Explain how this feature was tested and what kind of testing (unit, integration, UI) has been done. Mention platforms or configurations used for testing.
- **Manual Testing:**
    - [X] Tested on emulator / physical device.
- **Unit Tests**
    - [X] Added units tests for this

If you haven't added tests, please create a new issue and assign it.

##  Related Issues/Tickets

Closes #215.

## Screenshots (if applicable)

Screenshot 2024-11-26 at 16 32 57
Screenshot 2024-11-26 at 16 33 20

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt [hunk 1]
```diff
@@ -149,84 +167,61 @@ fun ProfileContent(
...
-          }
-    }
-  }
+      modifier =
+          Modifier.fillMaxWidth(0.80f)
+              .padding(horizontal = 10.dp, vertical = 8.dp)
+              .testTag("friendRequestCard")) {
+        Column(modifier = Modifier.padding(12.dp)) {
+          Text(
+              text = stringResource(R.string.pending_friend_requests, friendRequests.size),
```
Comment (g0g05arui) [inline app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt:184]: I think we should avoid using formatted strings as string resources. I understand the idea of not hard coding strings, but having a format like this is not always beneficial, I believe that the version before was a bit more readable.

 But, maybe if we decide to keep it like we can add multiple language support in the future, if we run out of features to code :)

**Review by g0g05arui — COMMENTED** [review]
Great job, the UI looks way better! Nothing much to add, it looks pretty straight forward

**Review by g0g05arui — APPROVED** [review]

File: app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt (hunk 1 above)
Comment (1andreea26) [reply app/src/main/java/com/android/voyageur/ui/profile/ProfileScreen.kt:184]: I'll keep this in mind for the future!

**Comment by 1andreea26** [comment]
> Great job, the UI looks way better! Nothing much to add, it looks pretty straight forward.

Thanks for the quick review! :)

**Pull request merged**